pkg/sidecar/pod.go

```diff
@@ -39,7 +40,11 @@ func add(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCo
 		return pod, err
 	}
 
-	container := collector.Container(cfg, logger, otelcol, false)
+	otc, err := convert.V1Alpha1to2(otelcol)
+	if err != nil {
+		return corev1.Pod{}, err
+	}
+	container := collector.Container(cfg, logger, otc, false)
```

Comment:
I think the replace config needs to come after the conversion `otelColCfg, err := collector.ReplaceConfig(otelcol)`